### PR TITLE
ES6: Convert async-call-merger.spec.js

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `async-call-merger.spec.js` to ES6.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/async-call-merger.spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    });
+
+    test('should call both callbacks with result of target callback', () => {
+        const callback1 = jest.fn();
+        targetFnWithMerging(callback1);
+
+        const callback2 = jest.fn();
+        targetFnWithMerging(callback2);
+
+        targetFn.mock.calls[0][0]('target result');
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/async-call-merger.spec.js]: It's not clear to me what this line is doing. Would it be clearer / accurate to use `[targetFn.mockReturnValueOnce('target result')](http://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockreturnvalueoncevalue)` at the top of this test?

Comment (gustavpursche) [reply static/src/javascripts/projects/common/modules/async-call-merger.spec.js]: In the old code this line was:

```js
targetFn.getCall(0).args[0]('target result');
```

That means: get the first argument, which got passed in the first call of `targetFn` and call it with `'target result'`. I have no idea how to properly express this in jest, but what I did is a [documented way of doing it](http://facebook.github.io/jest/docs/en/mock-functions.html).

Comment (SiAdcock) [reply static/src/javascripts/projects/common/modules/async-call-merger.spec.js]: I think so. I'm concerned that this test is tightly coupled to the implementation of the module. As such, it demonstrates (in a slightly mind-binding way) how the module works internally without showing us how to use it. I know you didn't introduce this problem.

I would change this by updating the `targetFn` definition in the `beforeEach` to this:

```js
        targetFn = jest.fn(callback => {
            callback('target result');
        });
```
This effectively replaces the line that reaches into the `targetFn` args and calls the internal `targetCallbackHandler`.

We can also get rid of the final test as this test will cover everything.

Comment (SiAdcock) [reply static/src/javascripts/projects/common/modules/async-call-merger.spec.js]: Even more unfortunately, this module is used in one place and can probably easily be replaced with a promise-based interface 🙄

**Comment by gustavpursche** [comment]
@SiAdcock Thanks for the great suggestions ❤️

**Review by SiAdcock — APPROVED** [review]
Thanks for the file rename too!

**Pull request merged**